src/hooks/use-assistant.ts

```diff
@@ -191,7 +192,7 @@ export const useAssistant = ( instanceId: string ) => {
 	}, [ instanceId ] );
 
 	return {
-		messages: messagesDict[ instanceId ] || [],
+		messages: messagesDict[ instanceId ] || EMPTY_MESSAGES,
```

Comment:
Nice fix! So now, if messages are empty, it will compare the same array defined as constant, so the comparison result will be `true`. In the case of the new empty array, the array object was different each time:

```
[] === []
false
EMPTY_MESSAGES === EMPTY_MESSAGES;
true
```

Reply:
Yes, exactly. React uses shallow comparison for props, so we should avoid re-creating objects/arrays if they have the same value. Otherwise, this pattern will result in extra re-renders ([ref](https://react.dev/reference/react/memo#my-component-rerenders-when-a-prop-is-an-object-or-array)).

When debugging re-renders with React DevTools, I realized that in an empty conversation, on every key event `AuthenticatedView` was being re-rendered due to the `messages` prop. However, the value reminded `[]`.